File: lib/levels/levels.dart
import 'package:batufo/levels/test/simple.dart';
import 'package:batufo/levels/tilemap.dart';

const Map<String, String> levels = {'simple': Levels.simple};

class Levels {
  static const simple = simpleLevel;

  static Tilemap tilemapForLevel(String levelName) {
    assert(levels.containsKey(levelName), 'cannot find level $levelName');
    return Tilemap.build(levels[levelName]);
  }
}

File: server/src/arena/levels.ts
import { strict as assert } from 'assert'
import largeLevel from './levels/large'
import simpleLevel from './levels/simple'
import crowdedLevel from './levels/crowded'
import singlePlayerLevel from './levels/single-player'
import { Tilemap } from './tilemap'
import { Level } from './level'

export class Levels {
  static tilemapForLevel(levelName: string): Tilemap {
    const level = levels.get(levelName)
    assert(level != null, 'cannot find level $levelName')
    return Tilemap.build(level.levelString)
  }
}

export const levels: Map<string, Level> = new Map([
  [singlePlayerLevel.name, singlePlayerLevel],
  [simpleLevel.name, simpleLevel],
  [crowdedLevel.name, crowdedLevel],
  [largeLevel.name, largeLevel],
])

export const levelNames: string[] = Array.from(levels.keys())
export const validLevel = (levelName: string) => levels.has(levelName)
